src/ResourceManager/AnalysisServices/Commands.AnalysisServices.Dataplane/Commands/Synchronize-AzureASInstance.cs

```diff
@@ -0,0 +1,221 @@
+using Microsoft.Azure.Commands.AnalysisServices.Dataplane.Properties;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using System.Net;
+
+namespace Microsoft.Azure.Commands.AnalysisServices.Dataplane
+{
+    /// <summary>
+    /// Cmdlet to log into an Analysis Services environment
+    /// </summary>
+    [Cmdlet("Synchronize", "AzureAnalysisServicesInstance", SupportsShouldProcess = true)]
+    [Alias("Synchronize-AzureAsInstance")]
+    [OutputType(typeof(bool))]
+    public class SynchronizeAzureAzureAnalysisServer: AzurePSCmdlet
+    {
+        private string serverName;
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "Name of the Azure Analysis Services server to synchronize",
+            Position = 0,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public string Instance { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The list of databases that need to be replicated",
+            Position = 1,
+            ValueFromPipeline = true)]
+        [ValidateNotNullOrEmpty]
+        public string[] Databases { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public SwitchParameter PassThru { get; set; }
+
+        protected override IAzureContext DefaultContext
+        {
+            get
+            {
+                // Nothing to do with Azure Resource Management context
+                return null;
+            }
+        }
+
+        public IAsAzureHttpClient AsAzureHttpClient { get; private set; }
+
+        public ITokenCacheItemProvider TokenCacheItemProvider { get; private set; }
+
+        public SynchronizeAzureAzureAnalysisServer(): this(new AsAzureHttpClient(() => new HttpClient()), new TokenCacheItemProvider())
+        {
+        }
+
+        public SynchronizeAzureAzureAnalysisServer(IAsAzureHttpClient AsAzureHttpClient, ITokenCacheItemProvider TokenCacheItemProvider)
+        {
+            this.AsAzureHttpClient = AsAzureHttpClient;
+            this.TokenCacheItemProvider = TokenCacheItemProvider;
+        }
+
+        public async override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Instance, Resources.RestartingAnalysisServicesServer))
+            {
+                var context = AsAzureClientSession.Instance.Profile.Context;
+                AsAzureClientSession.Instance.Login(context, null);
+                string accessToken = this.TokenCacheItemProvider.GetTokenFromTokenCache(AsAzureClientSession.TokenCache, context.Account.UniqueId);
+
+                Uri syncBaseUri = new Uri(string.Format("{0}{1}{2}", Uri.UriSchemeHttps, Uri.SchemeDelimiter, context.Environment.Name));
+
+                var syncResults = await Task.WhenAll(Databases.Select(databaseName => SynchronizeDatabaseAsync(context, syncBaseUri, databaseName, accessToken)));
```

Comment:
Inspect each of the results for syncstate if the syncstate is failed, then writeError(syncresults)